npm/svelte/src/index.ts

```diff
@@ -0,0 +1,163 @@
+/// <reference types="cypress" />
+import {
+  injectStylesBeforeElement,
+  StyleOptions,
+  getContainerEl,
+  setupHooks,
+} from '@cypress/mount-utils'
+
+const DEFAULT_COMP_NAME = 'unknown'
+
+// when we mount a Svelte component, we add it to the global Cypress object
+// so here we extend the global Cypress namespace and its Cypress interface.
+// Svelte users are used to using $set and $destroy to trigger the Svelte reactivity engine.
+// We have parallels to this in Cypress.vueWrapper.
+declare global {
+  // eslint-disable-next-line no-redeclare
+  namespace Cypress {
+    interface Cypress {
+      // TODO, types
+      svelteComponent: any
+    }
+  }
+}
+
+const svelteComponentOptions = [
+  'accessors',
+  'anchor',
+  'props',
+  'hydrate',
+  'intro',
+  'context',
+] as const
+
+export type CyMountOptions = Partial<Record<(typeof svelteComponentOptions)[number], any>> &
+{ log?: boolean } &
+Partial<StyleOptions>
+
+Cypress.on('run:start', () => {
+  // `mount` is designed to work with component testing only.
+  // it assumes ROOT_SELECTOR exists, which is not the case in e2e.
+  // if the user registers a custom command that imports `cypress/svelte`,
+  // this event will be registered and cause an error when the user
+  // launches e2e (since it's common to use Cypress for both CT and E2E.
+  // https://github.com/cypress-io/cypress/issues/17438
+  if (Cypress.testingType !== 'component') {
+    return
+  }
+
+  Cypress.on('test:before:run', () => {
+    Cypress.svelteComponent?.$destroy()
+    const el = getContainerEl()
+
+    el.innerHTML = ''
+  })
+})
+
+// implementation
+export function mount (
+  Component: any,
+  options: CyMountOptions = {},
+) {
+  // TODO: get the real displayName and props from VTU shallowMount
+  const componentName = getComponentDisplayName(Component)
+
+  const message = `<${componentName} ... />`
+  let logInstance: Cypress.Log
+
+  // then wait for cypress to load
+  return cy.then(() => {
+    if (options.log !== false) {
+      logInstance = Cypress.log({
+        name: 'mount',
+        message: [message],
+      })
+    }
+
+    // @ts-ignore
+    const document: Document = cy.state('document')
+
+    const el = getContainerEl()
+
+    injectStylesBeforeElement(options, document, el)
+
+    const componentNode = document.createElement('div')
+
+    componentNode.id = '__cy_svelte_root'
+
+    el.append(componentNode)
+
+    // mount the component using VTU and return the component on Cypress.svelteComponent
+    const ComponentConstructor = Component.default || Component
+
+    const component = Cypress.svelteComponent = new ComponentConstructor({
+      target: componentNode,
+      options,
+    })
```

Comment:
> Hmm, the `props` aren't getting passed through even though we're matching the API for testing-library. I'm surprised by that.

Saw your talk on SvelteSociety.
There is no `options` property on the [`options` parameter of a svelte component constructor](https://svelte.dev/docs#run-time-client-side-component-api-creating-a-component).

I guess this should be:
```ts
    const component = Cypress.svelteComponent = new ComponentConstructor({
      ...options,
      target: componentNode,
    })
```

Reply:
_Thank you_!